tests/eeid_plugin/enc/enc.c

```diff
@@ -0,0 +1,237 @@
+#else
+    OE_TEST(r == OE_OK);
+
+    oe_eeid_t* eeid = mk_test_eeid();
+    size_t v_endorsements_size = 0;
+    header_t* v_endorsements = make_endorsements(eeid, &v_endorsements_size);
+
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            evidence_size,
+            (uint8_t*)v_endorsements,
+            v_endorsements_size,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_OK);
+
+    free(v_endorsements);
+    free(eeid);
+    free(endorsements);
+#endif
+
+    OE_TEST(oe_free_evidence(evidence) == OE_OK);
+    OE_TEST(oe_free_claims_list(claims, claims_length) == OE_OK);
+}
+
+static void _test_get_evidence_fail()
+{
+    printf("====== running _test_get_evidence_fail\n");
+
+    uint8_t* evidence;
+    size_t evidence_size;
+
+    // Test get_evidence when plugin is unregistered.
+    OE_TEST(oe_unregister_attester(&eeid_attester) == OE_OK);
+
+    OE_TEST(
+        oe_get_evidence(
+            &eeid_attester.base.format_id,
+            0,
+            NULL,
+            0,
+            NULL,
+            0,
+            &evidence,
+            &evidence_size,
+            NULL,
+            NULL) == OE_NOT_FOUND);
+
+    OE_TEST(oe_register_attester(&eeid_attester, NULL, 0) == OE_OK);
+}
+
+static void _test_verify_evidence_fail()
+{
+    printf("====== running _test_verify_evidence_fail\n");
+
+    uint8_t* evidence;
+    size_t evidence_size;
+    uint8_t* endorsements;
+    size_t endorsements_size;
+    oe_claim_t* claims;
+    size_t claims_length;
+
+    OE_TEST(
+        oe_get_evidence(
+            &eeid_attester.base.format_id,
+            0,
+            NULL,
+            0,
+            NULL,
+            0,
+            &evidence,
+            &evidence_size,
+            &endorsements,
+            &endorsements_size) == OE_OK);
+
+    // Test verify_evidence with wrong sizes
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            0,
+            endorsements,
+            endorsements_size,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_INVALID_PARAMETER);
+
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            evidence_size,
+            endorsements,
+            0,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_INVALID_PARAMETER);
+
+    // Test verify evidence when plugin is unregistered
+    OE_TEST(oe_unregister_verifier(&eeid_verifier) == OE_OK);
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            evidence_size,
+            endorsements,
+            endorsements_size,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_NOT_FOUND);
+    OE_TEST(oe_register_verifier(&eeid_verifier, NULL, 0) == OE_OK);
+
+    // Test verify when evidence / endorsement id don't match?
+    // Test faulty verifier when they don't have the right claims?
```

Comment:
are these TODO's?

Reply:
Evidence of "good intentions" :-) I'm planning to add more tests, but we don't need them now.